crates/gitbutler-repo/src/repository_ext.rs

```diff
@@ -634,3 +629,429 @@ impl GixRepositoryExt for gix::Repository {
         Ok(self)
     }
 }
+
+#[cfg(test)]
+mod test {
+    mod create_wd_tree {
```

Comment:
All tests here should move to integration test level, as they are testing a public method.

Reply:
I'm quite a fan of having tests in the same file as the implementation as it makes them very easy to find, but they do make up half the file now, so I'll move them out.

Reply:
I mean, at the end of the day, one can also create a module file from the `tests` module, to get it out of the way. Keeping test-code in the `src/` tree also isn't as bad if one doesn't have to optimize their size for release.

In the end, it's just me wanting to remove one source of API errors where one accidentally tests private parts - it's a kind of mistake that usually is discovered when just trying to use the API from any other of the various crates, so after all it's probably never an issue in practice.